### ISSUE TITLE
Adding semantic conventions for setting attributes to record retry behavior

### ISSUE THREAD

**Issue by mponnambalam** [open-telemetry/opentelemetry-specification#729]
Adding semantic conventions for setting attributes to record retry behavior,

Esp. Long-running batch scenarios, a retry of a workflow from a specific step, is very common, and to differentiate which step is retried.  Retry details are very useful to diagnose and visual representation to differentiate the success path. 

This loosely can be achieved by setting attributes, standardizing with semantic convention will be of major help for consistency across implementations.

Proposal for to record  Retry of a Span , 

retry.id = Span Id for which currentSpan is being retried
retry.attempt =<count> 
retry.interval =100
retry.status =cancelled

**Comment by anuraaga** [comment]
Thanks for bringing this up @mponnambalam. One thing I've seen when instrumenting libraries is they often won't give a nice injection point for instrumenting each individual retry request. Concretely, this usually means not being able to set the headers, so it wouldn't be possible to assign different span IDs to each retry. So we'll need to decide whether we encourage a best effort of having spans for each retry, or otherwise fallback to something else like events; or, we could just encourage using events for all retries as a standard instead of spans. I'd personally lean towards the former but the latter could be considered more consistent.

**Comment by mponnambalam** [comment]
@anuraaga, good point on not always feasible to set a new span Id, hence looking for standardizing the convention. 

Every execution of code being retired enclosed in a span would surely have its previous span Id.  

To provide a background, ETL/Batch workflow engines, Oozie a job can be retried from any point of Job DAG. 

Event might not provide the desired correlation of metrics and resource utilization for each attempt.

Here are some Retry scenarios

1. Span is independent 
2. Span having child spans, which some or all of them had prior execution. 
3. Span may have a dependency, which some or all of them had prior execution.
4. combination of 2 & 3

In a workflow esp. batch, retry shall happen from any step, either from the start of trace or from a checkpoint. In such cases, some span shall have prior execution, might not be able to clearly distinguish its retry execution lineage, unless being as a new span with retry information.

Another thought, this can also be loosely attained utilizing links with attributes, similar to followsFrom relationship.

But still, needs clear semantics conventions for retry despite its fit into a link or at span, to differentiate retry path in a trace.

**Comment by ocelotl** [comment]
I asked Claude to review this issue and Claude suggests it be closed with this explanation:

> Purely a request to standardize retry semantic conventions (retry.id, retry.attempt, etc.); discussion fizzled in 2020 with no resolution and no cross-referencing PRs. All semantic-convention work migrated to open-telemetry/semantic-conventions, so any remaining interest in retry attributes should be tracked there, not in the spec repo.

To guard against a mistaken suggestion, Claude then independently re-verified this conclusion:

> OPEN, no stateReason; timeline has zero cross-refs/dupes/closing PRs; pure semconv ask (retry.id/attempt/speculative) with unresolved 2020 design debate. area:semantic-conventions belongs in open-telemetry/semantic-conventions, so moved-repo fits; caveat: no retry convention or tracking issue actually exists there yet (code search total_count=0), so the ask is unresolved, not truly migrated.

Claude evaluated this issue with medium confidence that it can be closed.

Claude has classified this issue as belonging in another repository.

The intention of this comment is to make it easier for a human being to review this issue before making a decision.

**Comment by svrnm** [comment]
@mponnambalam can you take a look at @ocelotl comment and make a call if this is the correct analysis and this can be closed? Thanks.
